pubsub/tests/unit/subscriber_test.py

```diff
@@ -282,11 +307,11 @@ async def test_producer_fetches_once_then_blocks(subscriber_client):
             )
         )
         await asyncio.sleep(0)
-        await asyncio.wait_for(queue.get(), 1)
+        await asyncio.wait_for(queue.get(), 1.0)
+        producer_task.cancel()
```

Comment:
What is this test trying to accomplish? It's not immediately clear to me. I think I'm not understanding the "then_blocks" part.

Reply:
It tests then consumer won't fetch new batch if consumer has not yet exhausted the queue. Probably worth renaming to something like `test_producer_fetches_once_then_waits_for_consumer`